<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Algo</class>
 <widget class="QWidget" name="Algo">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1081</width>
    <height>496</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Algo</string>
  </property>
  <layout class="QGridLayout" name="masterGrid" rowstretch="1,1,8">
   <item row="0" column="1">
    <widget class="QTextEdit" name="te_expl">
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="mouseTracking">
      <bool>false</bool>
     </property>
     <property name="inputMethodHints">
      <set>Qt::InputMethodHint::ImhNone</set>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
     </property>
     <property name="documentTitle">
      <string/>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
     <property name="tabStopDistance">
      <double>40.000000000000000</double>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::TextInteractionFlag::TextBrowserInteraction</set>
     </property>
     <property name="placeholderText">
      <string>OOPS - empty ...</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QGraphicsView" name="v">
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::SizeAdjustPolicy::AdjustToContents</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <property name="transformationAnchor">
      <enum>QGraphicsView::ViewportAnchor::AnchorUnderMouse</enum>
     </property>
     <property name="resizeAnchor">
      <enum>QGraphicsView::ViewportAnchor::AnchorUnderMouse</enum>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="buttonLayout">
     <item>
      <widget class="QToolButton" name="tb_expl">
       <property name="text">
        <string>show explanation</string>
       </property>
       <property name="icon">
        <iconset theme="QIcon::ThemeIcon::DialogInformation"/>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonStyle::ToolButtonTextBesideIcon</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="tb_hideExpl">
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset theme="document-revert"/>
       </property>
       <property name="iconSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="2,0,0">
     <item>
      <widget class="QGroupBox" name="gb_database">
       <property name="title">
        <string>Example-Data:</string>
       </property>
       <layout class="QGridLayout" name="gl_db" columnstretch="0,2,1">
        <item row="3" column="0">
         <widget class="QLabel" name="l_memory_blocks">
          <property name="text">
           <string>Memory Blocks:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QTextEdit" name="te_sizes"/>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="l_leftover_space">
          <property name="text">
           <string>Leftover-Space:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="l_chunk_sizes">
          <property name="text">
           <string>Data-Chunks (sizes):</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="sb_prio_cnt">
          <property name="suffix">
           <string> priorities</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>9</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
        <item row="3" column="1" colspan="2">
         <widget class="QLineEdit" name="le_pages"/>
        </item>
        <item row="4" column="1" colspan="2">
         <widget class="QSpinBox" name="sb_space">
          <property name="specialValueText">
           <string>auto (appended after last block)</string>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QPushButton" name="pb_simulate">
          <property name="text">
           <string>Simulate!</string>
          </property>
          <property name="icon">
           <iconset theme="QIcon::ThemeIcon::Computer"/>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="QTableWidget" name="tw_prio">
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
          </property>
          <attribute name="horizontalHeaderVisible">
           <bool>true</bool>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Priority</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Chunk-IDs</string>
           </property>
           <property name="textAlignment">
            <set>AlignLeading|AlignVCenter</set>
           </property>
          </column>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="l_algo">
          <property name="text">
           <string>Algorithm to use:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="cb_algo"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="gb_generate">
       <property name="title">
        <string>Auto-Generate:</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="4" column="2">
         <widget class="QSpinBox" name="sb_maxBcnt">
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="value">
           <number>32</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="2">
         <widget class="QSpinBox" name="sb_firstBlock">
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
          </property>
          <property name="suffix">
           <string>00</string>
          </property>
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>2</number>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="stepType">
           <enum>QAbstractSpinBox::StepType::AdaptiveDecimalStepType</enum>
          </property>
          <property name="value">
           <number>8</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="l_range_sizes">
          <property name="text">
           <string>Range of sizes:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QSpinBox" name="sb_extra">
          <property name="specialValueText">
           <string>auto</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="sb_minSz">
          <property name="layoutDirection">
           <enum>Qt::LayoutDirection::RightToLeft</enum>
          </property>
          <property name="accelerated">
           <bool>true</bool>
          </property>
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QPushButton" name="pb_generate">
          <property name="text">
           <string>Generate!</string>
          </property>
          <property name="icon">
           <iconset theme="process-working"/>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="sb_minBcnt">
          <property name="layoutDirection">
           <enum>Qt::LayoutDirection::RightToLeft</enum>
          </property>
          <property name="accelerated">
           <bool>true</bool>
          </property>
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="l_range_offsets">
          <property name="text">
           <string>Range of offsets:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="l_range_block_count">
          <property name="text">
           <string>Range of block count:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="sb_minBoffs">
          <property name="layoutDirection">
           <enum>Qt::LayoutDirection::RightToLeft</enum>
          </property>
          <property name="accelerated">
           <bool>true</bool>
          </property>
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="value">
           <number>12</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="sb_maxSz">
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="value">
           <number>200</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="l_first_block">
          <property name="text">
           <string>first Block:</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="l_extra_space">
          <property name="text">
           <string>Extra space (pages):</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QSpinBox" name="sb_maxBoffs">
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="value">
           <number>169</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="sb_maxCcnt">
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="value">
           <number>64</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="sb_minCcnt">
          <property name="layoutDirection">
           <enum>Qt::LayoutDirection::RightToLeft</enum>
          </property>
          <property name="accelerated">
           <bool>true</bool>
          </property>
          <property name="prefix">
           <string>$</string>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>256</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="l_range_chunk_count">
          <property name="text">
           <string>Range of chunk count:</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="gb_animation">
       <property name="title">
        <string>Animation:</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="2" column="1" colspan="2">
         <widget class="QRadioButton" name="rb_animated">
          <property name="text">
           <string>animated</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bg_visMode</string>
          </attribute>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QSlider" name="sl_aniDur">
          <property name="maximum">
           <number>2500</number>
          </property>
          <property name="singleStep">
           <number>50</number>
          </property>
          <property name="pageStep">
           <number>250</number>
          </property>
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TickPosition::TicksBelow</enum>
          </property>
          <property name="tickInterval">
           <number>500</number>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="l_anidur">
          <property name="frameShape">
           <enum>QFrame::Shape::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Shadow::Sunken</enum>
          </property>
          <property name="lineWidth">
           <number>5</number>
          </property>
          <property name="midLineWidth">
           <number>2</number>
          </property>
          <property name="text">
           <string>D
u
r.</string>
          </property>
          <property name="textFormat">
           <enum>Qt::TextFormat::PlainText</enum>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::TextInteractionFlag::NoTextInteraction</set>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QRadioButton" name="rb_silent">
          <property name="text">
           <string>silent</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bg_visMode</string>
          </attribute>
         </widget>
        </item>
        <item row="5" column="1" colspan="2">
         <widget class="QLCDNumber" name="lcd_aniDur">
          <property name="digitCount">
           <number>4</number>
          </property>
          <property name="segmentStyle">
           <enum>QLCDNumber::SegmentStyle::Flat</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="2">
         <widget class="QRadioButton" name="rb_visual">
          <property name="text">
           <string>visual</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bg_visMode</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>sl_aniDur</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lcd_aniDur</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1177</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>1153</x>
     <y>287</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="bg_visMode"/>
 </buttongroups>
</ui>
